src/TsTypes.ts

```diff
@@ -58,54 +61,99 @@ export namespace TsType {
   export interface TsProp {
     name: string
     required: boolean
-    type: TsType
+    type: TsTypeBase
   }
 
-  export class Any extends TsType {
+  export class Any extends TsTypeBase {
     _type() {
       return 'any'
     }
   }
-  export class String extends TsType {
+  export class String extends TsTypeBase {
     _type() {
       return 'string'
     }
   }
-  export class Boolean extends TsType {
+  export class Boolean extends TsTypeBase {
     _type() {
       return 'boolean'
     }
   }
-  export class Number extends TsType {
+  export class Number extends TsTypeBase {
     _type() {
       return 'number'
     }
   }
-  export class Object extends TsType {
+  export class Object extends TsTypeBase {
     _type() {
       return 'Object'
     }
   }
-  export class Void extends TsType {
+  export class Void extends TsTypeBase {
     _type() {
       return 'void'
     }
   }
-  export class Literal extends TsType {
+  export class Literal extends TsTypeBase {
     constructor(private value: any) { super() }
     _type() {
       return this.value
     }
   }
 
-  export class Array extends TsType {
-    constructor(private type?: TsType) { super() }
+  export class EnumValue {
+    identifier: string
+    value: string
+
+    constructor(enumValues: string[]) {
+      let hasValue = !!enumValues[0]
```

Comment:
Look at the call site for constructing these.  This is a zipped array; assuming we pass all other validation for an int enum, what are the possible outcomes here that we need to handle?  

I actually think most of this can probably go away... this was used to propagate string values in `enum: [...]` up to identifiers for enum declaration, which we no longer do.

```
var enumValues = zip(rule.tsEnumNames || [],
                // If we try to create a literal from an object, bad stuff can happen... so we have to toString it
                rule.enum!.map(_ => new TsType.Literal(_).toType(this.settings).toString()))
              .map(_ => new TsType.EnumValue(_))
```